### ISSUE TITLE
Warning: Failed prop type: Invalid prop `range` supplied to `VictoryBar`

### ISSUE THREAD

**Issue by peacechen** [FormidableLabs/victory-native#132]
This warning occurs with a horizontal stacked bar chart:
```Warning: Failed prop type: Invalid prop `range` supplied to `VictoryBar`.```

Code snippet:
```
<VictoryStack
	domain={{x:[0, 100]}}
	domainPadding={{y:5}}
	theme={this.state.chartTheme}
	width={this.state.chartWidth}
	height={sizes.h1}
	padding={0}
	colorScale={['blue', 'grey']}
	labels={() => null}
	horizontal={true}
>
	<VictoryBar data={[{x:name, y:percentage}]} />
	<VictoryBar data={[{x:name, y:100-percentage}]} />
</VictoryStack>
```

## Environment
Victory-native 0.14
iOS 10 simulator
React Native 0.45.1

**Comment by omarloren** [comment]
I can confirm this behavior, did you find a solution?

**Comment by peacechen** [comment]
It's still happening.  Doesn't seem to prevent the chart from rendering.  This bug is internal to Victory Charts / Native.  Do any of the maintainers have an idea why range is being passed to VictoryBar?

**Comment by hrmoller** [comment]
Just for the record: I get the same warning when using VictoryLine (non-native):

`Warning: Failed prop type: Invalid prop 'range' supplied to 'VictoryLine'.`
```js
<VictoryLine data={[{x: 0, y: 3}, {x: 2000, y: 3}]} style={{data: {stroke: '#dedede', strokeWidth: 1}}} />
```

**Comment by michaud** [comment]
I have the same warning when i make the chart small (width=100, height=100) but it goes away when I make it bigger (width=110, height=110)... It looks like the layout render code produces invalid values.

**Comment by boygirl** [comment]
fixed in latest versions

**Comment by jacargentina** [comment]
I'm having the same issue with latest version 0.27.2

checkPropTypes.js:19 Warning: Failed prop type: Invalid prop `range` supplied to `VictoryBar`

Using this:

```jsx
    <VictoryStack
      height={50}
      padding={{ bottom: 50 }}
      animate={{ onLoad: { duration: 800 } }}>
      {props.data.map((i, idx) => {
        return (
          <VictoryBar
            labelComponent={
              <VictoryLabel
                dy={-20}
                textAnchor="end"
              />
            }
            key={idx}
            style={{
              data: {
                fill: i.color,
                strokeWidth: 15
              },
              labels: {
                fontSize: 15
              }
            }}
            horizontal
            data={[
              {
                x: i.label,
                y: i.value,
                label: i.label
              }
            ]}
          />
        );
      })}
    </VictoryStack>
```

**Comment by JarLowrey** [comment]
I am also having this issue when range is specified on `VictoryAxis` or `VictoryChart` with latest `0.27.2` version

**Comment by peacechen** [comment]
This is still happening with the latest release 30.1.0 of victory-native.

Please re-open this issue.

**Comment by boygirl** [comment]
@peacechen sorry about that. I will attempt to reproduce.

**Comment by peacechen** [comment]
Thanks for looking into this again @boygirl

The code snippet in the OP recreates it.  Not sure if it's dependent on the width or height props.  The height I pass to it is small, approx value of 10.

**Comment by mmiocevic** [comment]
I know it is different, but similar issue happened when using VictoryGroup in VictoryChart in Victory (v 31).
Basically, component was throwing same error and whole chart was "broken" due to having
`height={this.state.chartHeight}`
and this chartHeight state property was smaller than this range prop.
For me solution was to increase height, e.g.
`height={this.state.chartHeight + data.length * 20}`

Basically try to tweak chart size (width and/or height).
Hope this helps.

@peacechen thank you for you last comment. It helped me realized for this issue.

Thanks!

**Comment by ablamunits** [comment]
I ran into this issue as well.
In my case I stored the initial (default) width of the chart in the component's state, and the initial value for the width was set to 0:

```
state = {
   width: 0
}

...

<VictoryChart width={this.state.width} />
```

This `width` is overridden on render, when the width of the container is calculated.
I was able to get rid of the errors by setting the initial width in the state to 1.

**Comment by vjsingh** [comment]
This is still happening for me when my VictoryChart was passed a height of 0 initially. Use case is using a hook to get the size of the window.
